### PR TITLE
SHRINKWRAP-515 Fix name and JDK5 compliance of AddPackage*Test

### PR DESCRIPTION
**Pull request by famod**
- suffix must be TestCase to be picked up by surefire
- removed JDK7 stuff
- consolidated by introducing test base class
- worked around file leak (?) on windows (just warn for now)

### CODE REVIEW

File: impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+public abstract class AddPackageTestBase {
+
+    /**
+     * Logger
+     */
+    private static final Logger LOG = Logger.getLogger(AddPackageTestBase.class.getName());
+
+    protected Domain domain;
+
+    private File tempFile;
```
Comment (bartoszmajsak) [inline impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java]: That's a very nice simplification, thanks for cleaning it up! I was wondering if we couldn't simply use [`TemporaryFolder` rule](http://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html) from JUnit to have this auto-cleanup done on our behalf?

File: impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+/**
+ * Base class for AddPackage* tests operating with a custom classloader.
+ *
+ * @author Falko Modler
+ * @author Bob McWhirter 
+ */
+public abstract class AddPackageTestBase {
+
+    /**
+     * Logger
```
Comment (bartoszmajsak) [inline impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java]: This is just using 3 lines for not that much value :) Let's kick it out :)

**Comment by bartoszmajsak** [comment]
Thanks for your contribution!

> removed JDK7 stuff

Can you be more specific on what has been changed? Is it `nio`?

File: impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java (hunk 1 above)
Comment (famod) [inline impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java]: Well, basically we could. But with the current file leak on windows (which prevents deletion anyway) there would not be a warning like with my current approach.

File: impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java (hunk 2 above)
Comment (famod) [inline impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java]: The `Logger` JavaDoc? Ok, will do. I just copied it from another class.

**Comment by famod** [comment]
> Can you be more specific on what has been changed? Is it `nio`?

Yes, and I had to replace `Collections.emptyEnumeration()` in `FilteringClassLoader`.

File: impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java (hunk 2 above)
Comment (bartoszmajsak) [inline impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java]: I know, but we can boy scout here - it's just a noise :)

File: impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java (hunk 1 above)
Comment (bartoszmajsak) [inline impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java]: Can you elaborate under which circumstances the leak occurs?

Comment (famod) [inline impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java]: No special circumstances really. I traced it back to the `.addPackage(...)` call in the test method. Once I remove this call (which renders the test pretty useless, though) the `.tearDown()` method does delete the file just fine.
In the beginning I thought that the `URLClassLoader` needs to be closed explicitly but firstly this is a JDK7+ feature and secondly this does not change anything (tried it briefly with JDK5_HOME pointing to JDK8).

Actually I was going to file another bug ticket anyway. WDYT?

Comment (bartoszmajsak) [inline impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java]: Please do so, but this makes me wonder what locks this file though. Do you have Linux to check if we are facing the same issue there?

Comment (famod) [inline impl-base/src/test/java/org/jboss/shrinkwrap/impl/base/spec/AddPackageTestBase.java]: I just tested this on CentOS7 (via VirtualBox) and the problem does _not_ occur there!

It is interesting to see that the problem _does_ occur on Linux when the project resides on a virtualbox shared folder (which is a Windows folder "physically") but there is no problem when the project resides entirely within the Linux VM.

**Comment by famod** [comment]
@bartoszmajsak I removed the Logger JavaDoc. I suggest we take another stab at `TemporaryFolder` while working on SHRINKWRAP-517 (the file leak).
AFAICS this PR should be now mergable.

**Comment by bartoszmajsak** [comment]
>  I suggest we take another stab at TemporaryFolder while working on https://issues.jboss.org/browse/SHRINKWRAP-517 (the file leak).

That makes perfect sense. Thanks for your contribution.

**Pull request merged**